server/src/main/java/org/elasticsearch/action/admin/indices/segments/IndicesSegmentResponse.java

```diff
@@ -104,6 +104,8 @@ public void writeTo(StreamOutput out) throws IOException {
 
     @Override
     public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        builder.startObject();
+        RestActions.buildBroadcastShardsHeader(builder, params, this);
```

Comment:
given that in most cases we have the shards header, how about doing this in `BroadcastResponse`:

```
public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
    builder.startObject();
    RestActions.buildBroadcastShardsHeader(builder, params, this);
    addCustomFields();
    builder.endObject();
}

protected void addCustomFields() {

}
```

subclasses can override this additional method to add their own fields rather than repeating the shards header every time. There is always the possibility to also override toXContent for content where the shards header is not printed out.